### ISSUE TITLE
involves_at_most_one_variable is not exported

### ISSUE THREAD

**Issue by wbhart** [Nemocas/AbstractAlgebra.jl#729]
Terrible name, but if the function is going to exist, it should be accessible.

**Comment by wbhart** [comment]
Nah, this is a custom hack someone added. We should leave it unexported. What they really wanted is a function isunivariate for multivariate polynomials. But that name is ambiguous in multiple ways.

I propose we have the following:

* isunivariate(R) for a multivariate polynomial ring R tells you if the ring R has just one variable
* isunivariate(p) for a multivariate polynomial tells you if the polynomial involves at most one variable (the zero polynomial being considered univariate)
* check <: MPolyElem if you want to know something about the type

**Comment by thofma** [comment]
So if I understand correctly, this is about renaming `involves_at_most_one_variable` to `isunivariate`, right?

**Comment by wbhart** [comment]
I think so. I have to go back and check what the original rationale was for not naming it that (apart from the fact that at the time we wanted it to be implemented using iterators and the code wasn't doing that). If what I propose here addresses the issue we had there, then yes, I think perhaps we should rename it, then export it.

**Comment by wbhart** [comment]
I found the issue where we made the naming decision [1].

The objection does indeed seem to be simply that you don't know if the user is asking whether the polynomial involves at most one variable or whether it is actually in a univariate ring. My proposal above deals with that problem, so I think we should do that, especially now that the actual code is not completely broken (MPoly was a mess at the time, and we didn't even have proper iterators everywhere).

If there are no objections I will go ahead and change the name to isunivariate (with a deprecation for the person using it) and then we can export the symbol.

[1] https://github.com/Nemocas/AbstractAlgebra.jl/issues/131

**Comment by wbhart** [comment]
I can also add isunivariate(::MPolyRing) which will return true iff there is one variable in the ring.

**Comment by fieker** [comment]
On Wed, May 19, 2021 at 02:37:15PM -0700, wbhart wrote:
> I found the issue where we made the naming decision [1].
> 
> The objection does indeed seem to be simply that you don't know if the user is asking whether the polynomial involves at most one variable or whether it is actually in a univariate ring. My proposal above deals with that problem, so I think we should do that, especially now that the actual code is not completely broken (MPoly was a mess at the time, and we didn't even have proper iterators everywhere).
> 
> If there are no objections I will go ahead and change the name to isunivariate (with a deprecation for the person using it) and then we can export the symbol.

Go ahead - I am just privately amused: is wrote a version of
isunivariate which then was changed to the complicated name and is now
coming back...

no complaint, just mild amusement

**Comment by wbhart** [comment]
@fieker Your isunivariate returns a polynomial as well, which is very expensive. Mine does not.

Any change you can rename yours?

cc @thofma
